api-documentation/shacl.md

```diff
@@ -0,0 +1,181 @@
+
+[shacl]: https://www.w3.org/TR/shacl/
+[ucr]: https://www.w3.org/TR/shacl-ucr/
+
+### Announce SHACL in API Documentation
+
+An API which chooses to use SHACL in addition to the built-in `hydra:Class` `MUST` announce that fact in its API Documentation resource
+
+```json
+{
+  "@context": {
+    "hydra": "http://www.w3.org/ns/hydra/core#",
+    "requires": { "@id": "hydra:requires", "@type": "@id" }
+  },
+  "@id": "/api",
+  "@type": "hydra:ApiDocumentation",
+  "requires": "http://www.w3.org/ns/hydra/shacl"
+}
+```
+
+By adding the `requires` property with the `hydra/shacl` identifier the client will be prepared to find SHACL terms within the payloads coming from this API.
+
+### Describing request bodies
+
+The most obvious use for Shapes is to use them to annotate supported operations' payloads. They will simply be added alongside `hydra:Class` as objects of `hydra:expects`.
+
+```json
+{
+  "@context": "/api/context.jsonld",
+  "@id": "/api",
+  "@type": "hydra:ApiDocumentation",
+  "supportedClass": [{
+    "@id": "mov:Movies",
+    "supportedOperation": {
+      "@type": "schema:CreateAction",
+      "method": "POST",
+      "expects": ["mov:Movie", "mov:NewMovieShape"]
+    }
+  }, {
+    "@id": "mov:Movie",
+    "@type": "Class",
+    "supportedProperty": [{
+      "property": "schema:name"
+    }, {
+      "property": "mov:director"
+    }, {
+      "property": "schema:genre"
+    }]
+  }]
+}
+```
+
+By adding a second value to the `expects` annotation, the client will discover that the payload of this operation is expected to be not only a representation of a `mov:Movie` but also that the request body has to conform to the given SHACL Shape.
+
+{% hint style="info" %}
+Client which do not understand SHACL will ignore it and proceed with only the `hydra:Class` description.
+{% endhint %}
+
+For clarity the `mov:NewMovieShape` is presented separately as it would appear in the API Documentation resource or as its own individual resource.
+
+```json
+{
+  "@id": "mov:NewMovieShape",
+  "@type": "sh:NodeShape",
+  "sh:targetClass": "mov:Movie",
+  "sh:property": [{
+    "sh:path": "schema:name",
+    "sh:or": [{
+      "sh:datatype": "xsd:string"
+    }, {
+      "sh:datatype": "rdf:langString"
+    }],
+    "sh:name": "Title",
+    "sh:description": "Movie's title",
+    "sh:minCount": 1
+  }, {
+    "sh:path": "mov:director",
+    "sh:class": "schema:Person",
+    "sh:nodeKind": "sh:IRI",
+    "sh:maxCount": 1
+  }, {
+    "sh:path": "schema:genre",
+    "sh:in": [
+      "Comedy",
+      "Drama",
+      "Documentary"
+    ]
+  }]
+}
+```
+
+While the Shape repeats some of the information typically found on the supported class `mov:Movie` it is clear that it can also provide much more information about the expected payload of the request to create a movie resource, which would be otherwise impossible with just Hydra Core terms:
```

Comment:
Not really - hydra's supportedProperty says more that client could perform some operations on those properties, they are somehow known and could be used, i.e. in Iri template mappings. This has nothing to do with data structure description

Reply:
Um, no? Otherwise what would be the use for `hydra:required` for instance? :shrug:

`Class` + `supportedProperty` totally define a very crude data model. SHACL can extend it with more information

Reply:
Ok, maybe I was too hasty with that opinion